tests/suites/cli/active_branch.sh

```diff
@@ -0,0 +1,88 @@
+# Checks juju indicates the correct branch output tabular and json.
+run_indicate_active_branch_no_active() {
+  echo
+
+  file="${TEST_DIR}/indicate-active-branch-no-active.txt"
+
+  ensure "indicate-active-branch-no-active" "${file}"
+
+  export JUJU_DEV_FEATURE_FLAGS=generations
+  juju deploy ubuntu 2>&1
+
+  wait_for "ubuntu" ".applications | keys[0]"
+  STATUS_NO_BRANCHES=$(juju status)
+  STATUS_JQ_NO_BRANCHES=$(juju status --format=json | jq '.branches')
+  check_not_contains "${STATUS_NO_BRANCHES}" "Branch"
+
+  if [ "${STATUS_JQ_NO_BRANCHES}" != null ]; then
+    echo "The status shows branches even though we do not use them yet"
+    exit 1
+  fi
+
+  destroy_model "indicate-active-branch-no-active"
+}
+
+run_indicate_active_branch_active() {
+  echo
+
+  file="${TEST_DIR}/indicate-active-branch-active.txt"
+
+  ensure "indicate-active-branch-active" "${file}"
+
+  export JUJU_DEV_FEATURE_FLAGS=generations
+  juju deploy ubuntu 2>&1
+
+  juju add-branch bla
+  wait_for "ubuntu" ".applications | keys[0]"
```

Comment:
Can we change this wait_for condition and wait for the idle condition. Then we can remove the `sleep 2` which won't be reliable on the CI infrastructure.

```suggestion
  wait_for "ubuntu" "$(idle_condition "ubuntu")"
```

The above is the best way to wait for an application to be deployed.